### ISSUE TITLE
Landing page affordance

### ISSUE THREAD

**Issue by scotthmurray** [processing/Processing-Hour-Of-Code#20]
While I love the current "Hello" landing page, I want to make sure it's clear that the "Hello" is clickable. We either need a button that explicitly says "Get started >" or similar, or some visual treatment that indicates "Hello" is how one gets started.

This is not high priority right now; I just wanted to log it here while I'm thinking about it.

(Let me know if this is too detailed to list as a separate issue.)

**Comment by scottgarner** [comment]
I definitely agree that we need some guidance. I was thinking that maybe after the Hello comes up, we get some additional copy beneath, maybe just a single line of instruction. I'll get something there in the next update and we can decide if text or a button is the best approach.

**Comment by scotthmurray** [comment]
Yes, perfect, like the "Hello" appears, and then a moment later, the single line cue appears, e.g. "Ready to get started?" as a link, or "Get started" or similar.

In a related idea, what if the "Hello" screen actually employed Processing to offer a teaser of what students are about to learn?  I'll mock up what I'm picturing and share it here.

**Comment by scotthmurray** [comment]
What do you all think of something like this?

http://alignedleft.com/clients/hourofcode/

The idea is to indicate motion and activity using things they will learn how to do in the course (draw rects and ellipses).  The colors are the IDE editor colors, so hopefully it feels "Processing-y".

Obviously the font isn't right in this sketch, but what do you think of the overall idea?

**Comment by scottgarner** [comment]
I'm not personally a fan of a continuous animation like this and both the color and movement are too busy for my taste.

That said, I think the idea of showing off Processing from the get-go is a good one. My thought would be to reveal the text through a Processing.js animation and perhaps to have it reactive to the mouse after it settles. Imagine, for example, a particle system with some basic physics to avoid the pointer. I think palette-wise, it would be nice to stick to the blues and yellow of Casey's earlier examples, though if those examples get more complex and introduce more colors, I think we could pull those as well.

The banner on the (dat.GUI documentation pages)[http://workshop.chromeexperiments.com/examples/gui/#1--Basic-Usage] isn't visually what I'd go for, but it's right on the mark insofar as it acts as a headline and a technology demo.

I'd have to think a bit about the best way to bring in the type beyond just cutting it up. It might be interesting to use the letterforms as a mask with the particle animation occurring behind it.

**Comment by scotthmurray** [comment]
Good ideas, I'll experiment more with the sketch.

**Comment by REAS** [comment]
Scott, instead of reinventing the wheel, what about using something already feels like the graphics in the main Processing website and in the IDE? 

We're making this for CSEd week, but I think people will use this tutorial as a general Processing introduction and it will be nice to have it graphically integrated.

For example, like on the Processing.js homepage. That code is here:
http://processingjs.org/sketches/bannerSketch.pde

I just started to make modifications here:
https://gist.github.com/REAS/7361591

Some ideas... it will need a light color background, we could use colors from the IDE text, and /or colors from the examples we're working with now.

**Comment by REAS** [comment]
Additional thoughts, the alpha of each line could go down toward the edge so it's not "in a box" or the whole thing can switch to inside a circle. I think the circle idea is better.

**Comment by scotthmurray** [comment]
Yes, I was thinking something along the same lines, and definitely using a circular form to avoid boxiness.  (Is boxiness a word?)

Working on a revised sketch and will share this later today.

**Comment by scotthmurray** [comment]
See this version, using the same basic code, but some adjustments to color, size, and alpha decreasing as elements move away from the center:

http://alignedleft.com/clients/hourofcode/v2/

**Comment by REAS** [comment]
I like it! And I like the soft palette. Can you try a yellow (255, 204, 0, alpha) rollover instead of red and see how that feels? 

What happens on this page? Should it say "Hello Processing" and "Click to start" or what? We have the license to use the Thesis fonts on Processing.org so we should use them here to integrate with the main site.

Maybe we can bring that palette into the main site. As mentioned elsewhere, Ben wasn't fond of the baby blue background and I put it there without conviction.

**Comment by scottgarner** [comment]
I'd caution against taking this palette into the site at this point since I do think we're going to need to work backwards from colors in the video. I tend to like subdued colors like this, too, but I have a hard time imagining them working well with brick, etc.

I think a text banner with a button/link for "click to start" makes sense. The form of "HELLO PROCESSING" would be rather long and thin, so it would have a different effect than the current center-surrounded text.

Scott, you're set as a Collaborator on this repository, so you can work with the actual splash page, which already has the fonts in the stylesheet. The pde file should go somewhere like /assets/pde/splash/hello.pde in the gh-pages branch and everything else in the root index.html

I have to say that I'd prefer to see something more dynamic/reactive as the intro, but understand if this is easiest on a condensed timeline.

**Comment by scotthmurray** [comment]
@scottgarner I'll take a look at the splash page code, try integrating this PJS piece, and let you know if I have any questions about the page structure.

**Comment by scottgarner** [comment]
I thought it might be weird to navigate an unfamiliar structure, so I quickly pulled your sketch and added it to the repository. A couple of notes:

1) The file is in /assets/pde/hello_processing/hello_processing.pde

2) I turned the text off so we can handle it with HTML and CSS, which will make it easier to style consistently and attach events. I think the button should be HTML and CSS, too. I can add that in an upcoming update.

3) I switched to a transparent background, again to make it easy to tweak CSS.

If you haven't used [jekyll](http://jekyllrb.com/docs/installation/) before, it's really easy to launch a local server for testing:

`jekyll server --watch`

**Comment by scottgarner** [comment]
I just pushed a few more tweaks to this with formatting changes, the start button and the full text.

Right now the button is just the default Bootstrap button and needs styling. I'm feeling better about this overall, though. I think once we have a new palette from the videos this page will be in good shape.

Some part of me really wants it to be "Hello, Processing", but I'm willing to ignore that impulse.

**Comment by scotthmurray** [comment]
Thanks, Scott. I tried getting jekyll set up, but am having install issues.  In any case…

1) Could you add hover styling for both the title "Hello Processing" and the "click to begin" text? On hover, they could turn pink (#E56498), for example.

2) For the title, I liked the simplicity of the original "Hello" but would also support "Hello, Processing" (with a comma) or "Hello, Processing." (with a comma and period).  But I'd defer to @reas and @shiffman.

3) To me, "click to begin" means I can click anywhere. I think we should rephrase this to indicate you have to click here, on the text, to begin.  How about "Let’s get started" or "Let’s get started &rarr;".  Let's also try underlining the text, so it looks like a clickable link, even before you mouse over it.

**Comment by scottgarner** [comment]
I think I was able to install pretty seamlessly with gem, but it's been a while. I think it was just: `gem install jekyll` I'm using the [homebrew](http://brew.sh/) version of ruby.

1) The banner currently isn't a link. I'm little torn on that. I kind of like the interaction of being able to play with the circles through it—as a link the mouse gets a bit confused.

I added some hover styling to the CSS for the button just so it's there, though I don't think we should worry about specific colors until we settle on a palette after reviewing the final videos.

2) I liked "Hello", too, though this is growing on me. I'm flexible on it.

3) You're right, I originally had it on a Bootstrap button where it made more sense. I think the floating text looks better than a button. As with the banner, I'm open to whatever copy @REAS and @shiffman prefer.

I'd prefer not to underline—I think we're in an application UI context on a page like this rather than a hypertext body where underlined links are expected/necessary.

We do have the option to have "Click to Begin" actually work when you click anywhere, too.

**Comment by scotthmurray** [comment]
Well, I'm OK without underlining, but I want to make sure the text looks clickable. Right now, to me, it doesn't.  I can rope in a couple people for some quick user testing and see how many people understand how to proceed.

**Comment by scottgarner** [comment]
Hah, I don't believe in user testing.

The text explicitly says "click here" and the pointer and color change on hover. Perhaps a case could be made to underline on hover, but if you can't figure out what to click without underlining in a default state, you won't even be able to make it past the [Hour of Code](http://csedweek.org) homepage, which has no underlines without mouse over.

**Comment by scottgarner** [comment]
Disclaimer: Please read the above with levity! I forget that comments don't always convey tone well.

Of course I'll defer to the group on this and of course it should look clickable, I just have unusually strong feelings about underlines in particular.

**Comment by scottgarner** [comment]
Closing this bug as the landing page is functionally complete. Restyling will occur as part of #1.
